src/beneuro_data/conversion/convert_nwb_to_pyaldata.py

```diff
@@ -0,0 +1,815 @@
+"""
+Module for conversion from nwb to pyaldata format
+"""
+
+import warnings
+
+import numpy as np
+import pandas as pd
+import scipy
+
+from ndx_pose import PoseEstimationSeries
+
+from pynwb import NWBHDF5IO
+from pynwb.behavior import SpatialSeries
+from pynwb.misc import Units
+
+
+def _bin_spikes(probe_units: Units, bin_size: float) -> np.array:
+    """
+    Bin spikes from pynwb from one probe
+
+    Parameters
+    ----------
+    probe_units :
+        PyNWB Units object from which to read spike times
+    bin_size :
+        Bin size in seconds to use
+
+    Returns
+    -------
+    Array :
+        Matrix of neuron x time with binned spikes
+
+    """
+
+    start_time = 0  # This is hardcoded since its aligned in nwb conversion
+    end_time = np.max(probe_units.spike_times[:])
+    number_of_bins = int(np.ceil((end_time - start_time) / bin_size))
+
+    # Initialize the binned spikes array
+    binned_spikes = np.zeros((len(probe_units.id[:]), number_of_bins), dtype=int)
+
+    # Populate the binned spikes array
+    for neuron_id in probe_units.id:
+        spike_times = probe_units.get_unit_spike_times(neuron_id)
+
+        # Taks spikes from after start time
+        spike_times = spike_times[spike_times >= start_time]
+
+        # Bin the spikes with flooring round
+        bin_indices = (spike_times / bin_size).astype(int)
+
+        # Add them to the corresponding bin.
+        np.add.at(binned_spikes, (neuron_id, bin_indices), 1)
```

Comment:
How about just using `np.histogram`?

Reply:
Mmm i kind of like np.add.at, feels more readable